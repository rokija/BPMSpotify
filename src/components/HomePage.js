import React, { Component } from 'react';
import { Link } from 'react-router-dom';
// import PropTypes from 'prop-types';
// import Cookies from 'universal-cookie';
// const cookies = new Cookies();

class HomePage extends Component {

    // componentWillMount() {
    //     const tokenData = cookies.get('token');
    //
    //     if (tokenData) {
    //         return true;
    //     }
    //     else {
    //         this.context.router.history.push("/login");
    //     }
    // }

    render() {
        return (
            <div className="home-page-wrapper">
                <div className="link-to-search-page-wrapper">
                    <h3 className="home-page-title">Get song data from spotify</h3>
                    <Link className="link-to-search-page" to="/search"><span>Enter</span></Link>
                </div>
            </div>
        );
    }
}


// HomePage.contextTypes = {
//     router: PropTypes.object
// };

export default HomePage;


