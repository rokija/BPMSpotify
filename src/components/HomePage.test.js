import React from 'react';
import { shallow } from 'enzyme';
import HomePage from './HomePage';
import toJson from 'enzyme-to-json';
import Cookies from 'universal-cookie';

const cookies = new Cookies();

const context = { router: { history: [] } };

describe('HomePage', () => {
    it('renders properly if token is set', () => {
        cookies.set('token', 'dfsaasd', { path: '/' });
        const wrapper = shallow(<HomePage />, { context });
        expect(wrapper.length).toEqual(1);
    });

    it('renders properly if token is not set', () => {
        cookies.set('token', '', { path: '/' });
        const wrapper = shallow(<HomePage />, { context });
        expect(wrapper.length).toEqual(1);
    });

    it('matches snapshot', () => {
        const wrapper = shallow(<HomePage />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
