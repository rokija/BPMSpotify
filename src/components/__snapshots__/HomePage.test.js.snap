// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage matches snapshot 1`] = `
<div
  className="home-page-wrapper"
>
  <div
    className="link-to-search-page-wrapper"
  >
    <h3
      className="home-page-title"
    >
      Get song data from spotify
    </h3>
    <Link
      className="link-to-search-page"
      replace={false}
      to="/search"
    >
      Enter
    </Link>
  </div>
</div>
`;
